dsc/src/main.rs

```diff
@@ -0,0 +1,518 @@
+        let uuid = format!("12345678-{0}-{0}-{0}-00000000{0}", port);
+        let start = Instant::now();
+        let output = Command::new(self.rs.ds_bin.clone())
+            .args(&[
+                "create",
+                "-d",
+                &new_region_dir,
+                "--uuid",
+                &uuid,
+                "--extent-count",
+                &extent_count,
+                "--extent-size",
+                &extent_size,
+                "--block-size",
+                &block_size,
+            ])
+            .output()
+            .unwrap();
+
+        let end = start.elapsed();
+        let time_f = end.as_secs() as f32 + (end.subsec_nanos() as f32 / 1e9);
+
+        if !output.status.success() {
+            println!(
+                "Create failed for {:?} {:?}",
+                self.rs.region_dir, output.status
+            );
+            println!(
+                "dir:{} uuid: {} es:{} ec:{}",
+                new_region_dir, uuid, extent_size, extent_count,
+            );
+            println!("Output:\n{}", String::from_utf8(output.stdout).unwrap());
+            println!("Error:\n{}", String::from_utf8(output.stderr).unwrap());
+            bail!("Creating region failed");
+        } else if !quiet {
+            println!(
+                "Downstairs region created at {} in {:04}",
+                new_region_dir, time_f,
+            );
+        }
+
+        let output_file = format!("downstairs-{}.txt", port);
+        let output_path = {
+            let mut t = self.output_dir.clone();
+            t.push(output_file);
+            t
+        };
+
+        let dsi = DownstairsInfo {
+            ds_bin: self.rs.ds_bin.clone(),
+            region_dir: new_region_dir,
+            port,
+            _create_output: String::from_utf8(output.stdout).unwrap(),
+            output_file: output_path,
+        };
+        self.rs.ds.push(dsi);
+        Ok(time_f)
+    }
+
+    /**
+     * Delete a region directory at the given port.
+     */
+    fn delete_ds_region(&mut self, port: u32) -> Result<()> {
+        // Create the path for this region by combining the region
+        // directory and the port this downstairs will use.
+        let mut my_region = PathBuf::new();
+        my_region.push(self.rs.region_dir.clone());
+        my_region.push(format!("{}", port));
+
+        let new_region_dir =
+            my_region.clone().into_os_string().into_string().unwrap();
+
+        std::fs::remove_dir_all(&new_region_dir)?;
+
+        // If this region was part of the ds vec, remove it.
+        self.rs.ds.retain(|ds| ds.port != port);
+        Ok(())
+    }
+
+    // Start all downstairs.
+    fn start_all_downstairs(&mut self) -> Result<()> {
+        for ds in self.rs.ds.iter() {
+            let cmd = ds.clone().start().unwrap();
+            println!("started ds: {:?}", cmd);
+            self.cmd.push(cmd);
+        }
+        Ok(())
+    }
+}
+
+// WIP.. begin ignore
```

Comment:
should this, and other XXX / ZZZ be addressed before merging?

Reply:
Alright.. alright.  
I was trying to sneak some in progress stuff in because I'm lazy, but, yeah, I'll
pull out the WIP stuff that does not even function yet, then after this goes in
put it back and continue working on it.